parsing/CSVBuilder.py

```diff
@@ -0,0 +1,68 @@
+import csv
+
+#This class will take a dictionary of strings and integers and transform it into a CSV file with the correct converted metric names.
+class CSVBuilder:
+    correctFormat = True
+    fname = ''
+    myDict = {}
+
+    #Constructor for the class.  Accept a filename as a string, and a dictionary.
+    #First, if the dictionary is not in the correct format (all keys are strings, and all values are integers),
+    #then set the correctFormat boolean to False, for when the build function is run.  Then, store the parameters
+    #in their corresponding global variables.
+    def __init__(self, filename: str, table: dict):
+        global correctFormat, fname, myDict
+        correctFormat = True
+        for x in table.keys():
+            if not isinstance(x, str):
+                correctFormat = False
+            if not isinstance(table[x], int):
+                correctFormat = False
+        fname = filename
+        myDict = table
+
+    #This function will build the CSV file and return an integer value.
+    #It will return 0 if it ran successfully, and 1 if it failed.
+    def build(self):
+        global myDict
+        cDict = {'NOL':'Number of Lines',
+                 'NOC':'Number of Comments',
+                 'CPES':'Ratio of comment Line to the executable statements',
+                 'NOCL':'Number of Classes',
+                 'NNC1':'Number of Classes of nesting level 1',
+                 'NNC2':'Number of Classes of nesting level 2',
+                 'NNC3':'Number of Classes of nesting level 3',
+                 'NNC4':'Number of Classes of nesting level 4',
+                 'NNC5':'Number of Classes of nesting level 5',
+                 'NNC6':'Number of Classes of nesting level 6',
+                 'NNC7':'Number of Classes of nesting level 7',
+                 'NNC8':'Number of Classes of nesting level 8',
+                 'NNC9':'Number of Classes of nesting level 9',
+                 'NNC10':'Number of Classes of nesting level 10',
+                 'NNC':'Numbre of Nested classes',
+                 'NOM':'Total Number of Methods',
+                 'NIM':'Number of Inherited Methods',
+                 'NLM':'Number of Local Methods',
+                 'NOVM':'Number of overridden methods',
+                 'ANIM':'Average Number of Inherited Methods per Class',
+                 'ANLM':'Average Number of Local methods per Class',
+                 'NOH':'Number of Hierarchies',
+                 'NAC':'Number of Abstract Classes',
+                 'NLC':'Number of Leaf Classes',
+                 'NPPM':'Average Number of Parameters per method',
+                 'NOA':'Number of Attributes',
+                 'AWI':'Average Width of Inheritance'}
```

Comment:
This could be a freestanding module-level contant outside of this class.

If for some reason we have to have multiple writers, there's no need for each of them to carry around a copy of this translation dictionary. In an strict OOP setting this would be an object with this as an attribute and no methods, or a constants class that doesn't get instantiated (no need for an object since these are compile time constants).